site/src/sphinx/advanced-saml.rst

```diff
@@ -1,6 +1,187 @@
+            .and()
+            .build();
+
+    Server server = new ServerBuilder()
+            .https(8443)
+            // Configure TLS with your key and certificate.
+            .tls(new File("your-certificate-file-path"), new File("your-key-file-path"))
+            // Decorate you service with SAML decorator.
+            .annotatedService("/", new MyService(), ssp.newSamlDecorator())
+            // Add SAML service to your server which handles a SAML response and a metadata request.
+            .service(ssp.newSamlService())
+            .build();
+
+How to handle the authentication response
+-----------------------------------------
+
+``armeria-saml`` provides :api:`SamlSingleSignOnHandler` to handle the response from an identity provider.
+It consists of ``loginSucceeded()`` and ``loginFailed()`` methods which handle the response,
+and ``beforeInitiatingSso()`` which handle a request. Most of cases, you may only need to write the two methods
+which handle the response, but if you want to send a data to your identity provider and get it back
+with a response, you need to implement ``beforeInitiatingSso()`` method.
+
+The following example shows a simple implementation of the :api:`SamlSingleSignOnHandler`. In the example,
+if an authentication is succeeded, an email address is retrieved from the response by referring to a ``name ID``
+element in the assertion, then it is sent to the end user via ``Set-Cookie`` header. It means that your
+:api:`Authorizer` can identify an authenticated session with a ``Cookie`` header in the following requests,
+like ``MyAuthorizer`` in the example.
+
+.. code-block:: java
+
+    class MySamlSingleSignOnHandler implements SamlSingleSignOnHandler {
+        @Override
+        public HttpResponse loginSucceeded(ServiceRequestContext ctx, AggregatedHttpMessage req,
+                                           MessageContext<Response> message, @Nullable String sessionIndex,
+                                           @Nullable String relayState) {
+            final Response response = message.getMessage();
+            final String username = response.getAssertions().stream()
+                                            .map(s -> s.getSubject().getNameID())
+                                            .filter(id -> id.getFormat().equals(SamlNameIdFormat.EMAIL.urn()))
+                                            .map(NameIDType::getValue)
+                                            .findFirst()
+                                            .orElse(null);
+            if (username == null) {
+                return HttpResponse.of(HttpStatus.OK, MediaType.HTML_UTF_8,
+                                       "<html><body>Username is not found.</body></html>");
+            }
+
+            final Cookie cookie = new DefaultCookie("username", username);
+            cookie.setHttpOnly(true);
+            cookie.setDomain("localhost");
+            cookie.setMaxAge(60);
+            cookie.setPath("/");
```

Comment:
We could mention that a user should put something more than just a username here to prevent tampering, perhaps mentioning JWT?